src/dotenv/main.py

```diff
@@ -206,7 +209,7 @@ def _replacement(name):
         then look into the dotenv variables
         """
         ret = os.getenv(name, new_values.get(name, ""))
-        return ret
+        return ret if ret is not None else name
```

Comment:
`ret` can't be `None` here, so I think the `if is not None` test shouldn't be added.  Also, returning `name` looks incorrect because `""` should be the default value.

It seems that Mypy detects an unrelated issue, but in that case I would suggest adding `# type: ignore`, if necessary, since it's probably a false positive from Mypy.

Reply:
Same `mypy` juggling here. I'll just ignore